src/backend/gporca/libgpopt/src/xforms/CXformUpdate2DML.cpp

```diff
@@ -144,13 +144,14 @@ CXformUpdate2DML::Transform(CXformContext *pxfctxt, CXformResult *pxfres,
 	// generate oid column and project operator
 	CExpression *pexprProject = NULL;
 	CColRef *pcrTableOid = NULL;
+	// seems this if statement should be changed, because at adb-6.x-dev plans - there are
+	// there are no info about tableoid, also slot at debug doesn't have this info
 	if (ptabdesc->IsPartitioned())
 	{
+		pcrTableOid = popUpdate->PcrTableOid();
```

Comment:
1  Deletion of [pcrOid generation at CXformUpdate2DML.cpp](https://github.com/arenadata/gpdb/pull/409/commits/05d6898c381640b9e6b13a9f99a9a09b89585458#diff-a44f4fd7c55d61e6605ae66bffda453c453e022468e3fbc443038adf1d3844a1) for partitioned tables (for update operation) and using the pcrOid from LogicalUpdate will broke some test cases from `CDMLTest`:
  - src/backend/gporca/data/dxl/minidump/UpdateDistKeyMismatchedDistribution.mdp
  - src/backend/gporca/data/dxl/minidump/UpdateDroppedCols.mdp
  - src/backend/gporca/data/dxl/minidump/UpdateNoDistKeyMismatchedDistribution.mdp
  - src/backend/gporca/data/dxl/minidump/UpdatePartTable.mdp

due to next [assertion]((https://github.com/arenadata/gpdb/blob/762fcb30a9e4a1f0d9af1ff4d9f06fa88e5dc35c/src/backend/gporca/libgpopt/src/xforms/CXformUpdate2DML.cpp#L171)) (test cases doesn't have OidCol at dxlLogicalUpdate node). These tests may be [fixed by adding OidCol](https://github.com/arenadata/gpdb/pull/409/commits/0c670de15ec2c87718b447e785b2a932cff86b9f) to the dxlLogicalUpdate node, but seems it will lead to problem with replay of plan generated on previous versions of gpdb (there is no OidCol so assert will fail). There is two approaches - remove the Assertion with fixing tests or generate pcrOid if Update node doesn't have prcOid.